TESTING.rst

```diff
@@ -474,20 +473,58 @@ More information:
 Airflow System Tests
 ====================
 
-The System tests for Airflow are not yet fully implemented. They are Work In Progress of the
-`AIP-4 Support for System Tests for external systems <https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-4+Support+for+System+Tests+for+external+systems>`__.
 These tests need to communicate with external services/systems that are available
 if you have appropriate credentials configured for your tests.
-The tests derive from ``tests.system_test_class.SystemTests`` class.
+The tests derive from ``tests.system_test_class.SystemTests`` class. They should also
+be marked with ``@pytest.marker.system(SYSTEM)`` where system designates the system
+to be tested (for example ``google.cloud``). Those tests are skipped by default.
+You can execute the tests providing ``--systems SYSTEMS`` flag to pytest.
 
-The system tests execute a specified
-example DAG file that runs the DAG end-to-end.
+The system tests execute a specified example DAG file that runs the DAG end-to-end.
+
+Most of the system tests have also addtional requirements for example they need to have
```

Comment:
I think it is very dangerous that this key is available in the container, and certainly not acceptable if we want to finish AIP-4. I have an idea that we can discuss tomorrow in the office. This will remove the need for key files in the container and even on the developer's computer.

Reply:
How this is dangerous for your local environment where you control both host and container?

Reply:
Note that this is only for local testing for now and manual test execution. We do not yet plan execution in automated way in CI and I am open to discussion here for sure.

Reply:
I agree however it should not be default - i will make it optional . I am also planning to have some small modifications to make it even easier to use with older versions of airflow. Bear with me,